cot-macros/src/lib.rs

```diff
@@ -192,3 +192,58 @@ pub(crate) fn cot_ident() -> proc_macro2::TokenStream {
         }
     }
 }
+
+#[proc_macro_derive(FromRequestParts)]
+pub fn derive_from_request_parts(input: TokenStream) -> TokenStream {
+    let ast = parse_macro_input!(input as syn::DeriveInput);
+    impl_from_request_parts_for_struct(&ast).into()
+}
+
+fn impl_from_request_parts_for_struct(ast: &syn::DeriveInput) -> proc_macro2::TokenStream {
+    let struct_name = &ast.ident;
+    let cot = cot_ident();
+
+    let fields = match &ast.data {
+        Data::Struct(data_struct) => match &data_struct.fields {
+            Fields::Named(fields_named) => &fields_named.named,
+            Fields::Unnamed(_) => {
+                let err = Error::custom(
+                    "Structs with unnamed fields are not supported for `FromRequestParts`",
+                );
+                return err.write_errors().into();
+            }
```

Comment:
I think we should support structs with unnamed fields as well. I wouldn't do this myself, but someone could create a context like `struct BaseContext(Urls, StaticFiles)` and I think it should be fine.